### PR TITLE
[MM-36432][MM-37072][MM-37073] Logic to support Focalboard and Playbooks tabs

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This PR contains the changes to support the newest version of the tabs with Playbooks and Focalboard. It requires some webapp changes to work properly, those will be implemented in version 6.0.

What's working:
- Moving and closing of Playbooks/Focalboard tabs
- Clicking on a link in one tab to bring you to another tab
- Re-opening of close tabs when a link is clicked.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36432
https://mattermost.atlassian.net/browse/MM-37072
https://mattermost.atlassian.net/browse/MM-37073

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: src/common/tabs/FocalboardTabView.ts
Comment (Willyfrog) [inline src/common/tabs/FocalboardTabView.ts:11]: this might change in the future, right? should we create a function that abstracts it? 0/5 though

File: src/common/tabs/PlaybooksTabView.ts
Comment (Willyfrog) [inline src/common/tabs/PlaybooksTabView.ts:11]: we might want to have a different team ;)

File: src/common/utils/url.ts
Comment (Willyfrog) [inline src/common/utils/url.ts:87]: why do we only make it lowercase if it ends with '/'?

File: src/main/Validator.ts
Comment (Willyfrog) [inline src/main/Validator.ts:105]: shouldn't this depend on whether it is focalboard or not?

File: src/main/windows/windowManager.ts
Comment (Willyfrog) [inline src/main/windows/windowManager.ts:481]: why do we need a while structure here? shouldn't this just run once?

File: src/main/windows/windowManager.ts
Comment (Willyfrog) [inline src/main/windows/windowManager.ts:503]: same as above, not really  sure why do we need the while loop

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Looking mostly good, a few small changes and comments

File: src/common/tabs/FocalboardTabView.ts
Comment (devinbinnie) [reply src/common/tabs/FocalboardTabView.ts:11]: This would be the only source of truth for the app. I figure if we ever allow for it to be modified per server, then maybe we'd have to add it to a configuration or something. But for now I think it works here.

File: src/common/utils/url.ts
Comment (devinbinnie) [reply src/common/utils/url.ts:87]: I copied this from above, not sure why it was done that way.
https://github.com/mattermost/desktop/pull/1680/files/940a6f4803317dda9ea7d18f671bba1040b2e25f#diff-abddc5347f7e36fa964b14961781c90b425e015942e189a1bd61f09d6105aad6L82

File: src/main/Validator.ts
Comment (devinbinnie) [reply src/main/Validator.ts:105]: I don't think that matters, we want to be able to close any tab theoretically, but we're currently just disallowing the closing of Messages. We want each tab type to be written such that it controls whether it's closable or not, but I don't think we need to limit that here, it should just be ignored if you tried to close a tab that isn't closable.

File: src/main/windows/windowManager.ts
Comment (devinbinnie) [reply src/main/windows/windowManager.ts:481]: `filteredTabs` is the list of open tabs. If `nextOrder` would normally pick a tab that's closed, `findIndex` would return -1 and it would error out. We want to make sure it's always picking the next tab.

File: src/main/windows/windowManager.ts
Comment (devinbinnie) [reply src/main/windows/windowManager.ts:503]: See above

File: src/main/Validator.ts
Comment (devinbinnie) [reply src/main/Validator.ts:105]: True, I don't think it does any harm.

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Nice work @devinbinnie. Looks like things are coming along! A few things I discovered:

1. When I click one of the `Messaging`, `Focalboard` or `Playbooks` tabs, the server dropdown flashes sometimes. This happens when I drag tabs to move them as well.


2. I think it's official now that we can change the product names:
  - Messaging -> Channels
  - Focalboard -> Boards
  - Playbooks (stays as is)

3. I'm not sure how to test "Clicking on a link in one tab to bring you to another tab"
  - It would appear that the focalboard button in the channel header does not open in the Focalboard tab
  - It would appear that the incident collaboration menu item from the main menu does not open the playbooks tab

4. Same as above. I can't seem to test "Re-opening of close tabs when a link is clicked."

File: src/common/utils/url.ts
Comment (Willyfrog) [reply src/common/utils/url.ts:87]: we should probably check for ending in `/` and make it lowercase in both places, most likely there is no problem with that, but if we ever run into a problem caused by that we are going to be soo confused 😆

**Review by Willyfrog — CHANGES_REQUESTED** [review]
minor change request (we can address that in a different PR if you prefer 0/5) otherwise LGTM

**Comment by matthewbirtch** [comment]
Hey @devinbinnie I'll be out this week, so if you need a UX review on this, please request a review from @andrewbrown00. Thanks!

File: .circleci/config.yml
Comment (Willyfrog) [inline .circleci/config.yml:452]: should we remove this before merging?

**Review by Willyfrog — APPROVED** [review]

File: .circleci/config.yml
Comment (devinbinnie) [reply .circleci/config.yml:452]: Probably. It won't affect anything after the PR is merged. I've been trying to look into a way to get signed builds made up when a label is added so that UX/QA can get test builds when they need them.

**Review by BenCookie95 — APPROVED** [review]
LGTM!

**Review by andrewbrown00 — CHANGES_REQUESTED** [review]
Hey, @devinbinnie, a few items to resolve:

- [ ] There's something funky going on with **Boards**, it's not displaying as its own tab. Instead, it just swaps with the open **Channels** tab and drops the mention, but the label stays "Channels".

   

- [ ] I still the **flash** issue Matt captured, when I click one of the Channels, Boards, or Playbooks tabs, the server dropdown flashes. This happens when I drag tabs to move them as well.
   
   

- [ ] Dragging the tabs to reorder is working, it just snaps the tabs back to their starting position

**Comment by devinbinnie** [comment]
Spoke to Juho, we'll do Security Review after merge.

**Review by andrewbrown00 — APPROVED** [review]
Thanks, @devinbinnie, changes look good 🙂 

Verified the following:
* Tabs drag and drop correctly ✅ 
* Clicking on links open the corresponding tabs ✅ 
   * Also happens when a tab has been closed ✅ 
* Channels tab can't be closed ✅ 
* Boards and Playbook tabs can be closed ✅ 

Dark mode | Light mode
----- | -----
image | image



**Notes** 
* Boards doesn't open when using the product switcher, this is a known issue https://github.com/mattermost/focalboard/issues/1142
* We are going to resolve the **badge count** alignment in a separate PR

**Pull request merged**
